libs/vkd3d/command.c

```diff
@@ -2211,11 +2202,15 @@ static void d3d12_command_list_transition_resource_to_initial_state(struct d3d12
     barrier.subresourceRange.baseArrayLayer = 0;
     barrier.subresourceRange.layerCount = VK_REMAINING_ARRAY_LAYERS;
 
+    vk_access_and_stage_flags_from_d3d12_resource_state(list->device, resource->initial_state,
+            list->vk_queue_flags, &dst_stage_mask, &barrier.dstAccessMask);
+
     TRACE("Initial state %#x transition for resource %p (old layout %#x, new layout %#x).\n",
             resource->initial_state, resource, barrier.oldLayout, barrier.newLayout);
 
-    VK_CALL(vkCmdPipelineBarrier(list->vk_command_buffer, src_stage_mask, dst_stage_mask, 0,
-            0, NULL, 0, NULL, 1, &barrier));
+    VK_CALL(vkCmdPipelineBarrier(list->vk_command_buffer,
+            VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT, dst_stage_mask,
```

Comment:
Can't the initial transition happen in TOP_OF_PIPE? Nothing on GPU should be able to access it before we make the transition.

Reply:
Yes, this was meant to be `TOP_OF_PIPE`, not `BOTTOM`.

Reply:
Fixed.